src/commands.rs

```diff
@@ -506,12 +506,16 @@ pub fn command_json_set<M: Manager>(
 
     while let Some(s) = args.next() {
         match s.try_as_str()? {
-            arg if arg.eq_ignore_ascii_case("NX") => set_option = SetOptions::NotExists,
-            arg if arg.eq_ignore_ascii_case("XX") => set_option = SetOptions::AlreadyExists,
+            arg if arg.eq_ignore_ascii_case("NX") && set_option == SetOptions::None => {
+                set_option = SetOptions::NotExists
+            }
+            arg if arg.eq_ignore_ascii_case("XX") && set_option == SetOptions::None => {
+                set_option = SetOptions::AlreadyExists
+            }
```

Comment:
`NX` and `XX` can now occur only once, but `FORMAT` can still occur more times, taking its last value?
(consider adding a `None` value also for FORMAT enum?)

Reply:
@oshadmi I didn't really deal with Format since we don't really support BSON yet, the all FORMAT thing needs a review